patchwork-registries/src/main/java/net/minecraftforge/registries/RegistryBuilder.java

```diff
@@ -0,0 +1,347 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.registries;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import org.apache.logging.log4j.LogManager;
+import com.google.common.collect.Lists;
+import net.minecraftforge.registries.IForgeRegistry.AddCallback;
+import net.minecraftforge.registries.IForgeRegistry.BakeCallback;
+import net.minecraftforge.registries.IForgeRegistry.ClearCallback;
+import net.minecraftforge.registries.IForgeRegistry.CreateCallback;
+import net.minecraftforge.registries.IForgeRegistry.DummyFactory;
+import net.minecraftforge.registries.IForgeRegistry.MissingFactory;
+import net.minecraftforge.registries.IForgeRegistry.ValidateCallback;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class RegistryBuilder<T extends IForgeRegistryEntry<T>> {
+	private static final int MAX_ID = Integer.MAX_VALUE - 1;
+
+	private Identifier registryName;
+	private Class<T> registryType;
+	private Identifier optionalDefaultKey;
+	private int minId = 0;
+	private int maxId = MAX_ID;
+	private List<AddCallback<T>> addCallback = Lists.newArrayList();
+	private List<ClearCallback<T>> clearCallback = Lists.newArrayList();
+	private List<CreateCallback<T>> createCallback = Lists.newArrayList();
+	private List<ValidateCallback<T>> validateCallback = Lists.newArrayList();
+	private List<BakeCallback<T>> bakeCallback = Lists.newArrayList();
+	private boolean saveToDisc = true;
+	private boolean sync = true;
+	private boolean allowOverrides = true;
```

Comment:
> override is not enabled by default, so this is fine
It appears that it is enabled by default. It needs to be disabled on vanilla registries for now.

Reply:
I see, gonna to fix that in the next commit